funsor/gaussian.py

```diff
@@ -317,7 +323,24 @@ def eager_reduce(self, op, reduced_vars):
             return result.reduce(ops.logaddexp, reduced_ints)
 
         elif op is ops.add:
-            raise NotImplementedError('TODO product-reduce along a plate dimension')
+            for v in reduced_vars:
+                if self.inputs[v].dtype == 'real':
+                    raise ValueError("Cannot sum along a real dimension: {}".format(repr(v)))
+
+            # Fuse Gaussians along a plate. Compare to eager_add_gaussian_gaussian().
+            old_ints = OrderedDict((k, v) for k, v in self.inputs.items() if v.dtype != 'real')
+            new_ints = OrderedDict((k, v) for k, v in old_ints.items() if k not in reduced_vars)
+            inputs = OrderedDict((k, v) for k, v in self.inputs.items() if k not in reduced_vars)
+
+            precision = Tensor(self.precision, old_ints).reduce(ops.add, reduced_vars)
+            precision_loc = Tensor(_mv(self.precision, self.loc),
+                                   old_ints).reduce(ops.add, reduced_vars)
+            loc = Tensor(_sym_solve_mv(precision.data, precision_loc.data), new_ints)
+            expanded_loc = align_tensor(old_ints, loc)
+            quadratic_term = Tensor(_vmv(self.precision, expanded_loc - self.loc),
```

Comment:
The math looks right.  Are `self.precision` and `self.loc` guaranteed to be aligned correctly so that these linear algebra helpers do the right thing?

Reply:
Yes, `align_tensor` should take care of that, but I'll throw in an assertion to make it clear:
- `self.precision`, `self.loc`, and `expanded_loc` are aligned to `old_ints`
- `loc` is aligned to `new_loc`